api/client/cloud/cloud_test.go

```diff
@@ -908,3 +922,25 @@ func (s *cloudSuite) TestAddCloudsCredentials(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(result, gc.DeepEquals, []params.UpdateCredentialResult{{}})
 }
+
+type cloudCredentialMatcher struct {
+	arg params.UpdateCredentialArgs
+}
+
+func (c cloudCredentialMatcher) Matches(x interface{}) bool {
+	cred, ok := x.(params.UpdateCredentialArgs)
+	if !ok {
+		return false
+	}
+	if len(cred.Credentials) != len(c.arg.Credentials) {
+		return false
+	}
+	if cred.Force != c.arg.Force {
+		return false
+	}
```

Comment:
you can also add checking of the c.arg.Credential contents.

Reply:
Sure thing.